### PR TITLE
Type unification should always use the canonical type

### PR DESCRIPTION
**Pull request by mbudiu-vmw**

### CODE REVIEW

**Comment by mbudiu-vmw** [comment]
Fixes #3364

File: backends/ebpf/psa/ebpfPsaTable.cpp [hunk 1]
```diff
@@ -690,7 +690,6 @@ void EBPFTablePSA::emitKeysAndValues(CodeBuilder *builder,
             CHECK_NULL(fieldName);
             builder->emitIndent();
             builder->appendFormat("%s.%s = ", keyName.c_str(), fieldName.c_str());
-            auto mtdecl = program->refMap->getDeclaration(keyElement->matchType->path, true);
```
Comment (jnfoster) [inline backends/ebpf/psa/ebpfPsaTable.cpp:693]: This seems unrelated?

**Review by jnfoster — APPROVED** [review]

File: backends/ebpf/psa/ebpfPsaTable.cpp (hunk 1 above)
Comment (mbudiu-vmw) [reply backends/ebpf/psa/ebpfPsaTable.cpp:693]: Yes, it is a separate commit. Another unused variable.
